src/iris_grib/_load_convert.py

```diff
@@ -1724,76 +1728,114 @@ def vertical_coords(section, metadata):
+        try:
+            typeOfFirstFixedSurface = section["typeOfFirstFixedSurface"]
+        except KeyError:
+            typeOfFirstFixedSurface = _TYPE_OF_FIXED_SURFACE_MISSING
+        try:
+            typeOfSecondFixedSurface = section["typeOfSecondFixedSurface"]
+        except KeyError:
+            typeOfSecondFixedSurface = _TYPE_OF_FIXED_SURFACE_MISSING
 
         # We treat fixed surface level type=1 as having no vertical coordinate.
         # See https://github.com/SciTools/iris/issues/519
-        if typeOfFirstFixedSurface not in [_TYPE_OF_FIXED_SURFACE_MISSING, 1]:
-            key = "scaledValueOfFirstFixedSurface"
-            scaledValueOfFirstFixedSurface = section[key]
-            if scaledValueOfFirstFixedSurface == _MDI:
-                if options.warn_on_unsupported:
-                    msg = (
-                        "Unable to translate type of first fixed "
-                        "surface with missing scaled value."
-                    )
-                    warnings.warn(msg)
+        if typeOfFirstFixedSurface in [
+            _TYPE_OF_FIXED_SURFACE_MISSING,
+            1,
+        ] and typeOfSecondFixedSurface in [_TYPE_OF_FIXED_SURFACE_MISSING, 1]:
+            return
+        fixed_surface_missing = FixedSurface(None, None, None, None)
+        fixed_surface_first = _FIXED_SURFACE.get(
+            typeOfFirstFixedSurface, fixed_surface_missing
+        )
+        if fixed_surface_first.point:
+            # We have a surface type that includes the point in its definition
+            first = fixed_surface_first.point
+        else:
+            first = _get_surface_value(section, "First", warn_only=True)
+            point = first
+
+        if typeOfSecondFixedSurface != _TYPE_OF_FIXED_SURFACE_MISSING:
+            fixed_surface_second = _FIXED_SURFACE.get(
+                typeOfSecondFixedSurface, fixed_surface_missing
+            )
+            if fixed_surface_second.point is not None:
+                # We have a surface type that includes the point in its definition
+                second = fixed_surface_second.point
             else:
-                fixed_surface_missing = FixedSurface(None, None, None)
-                fixed_surface = _FIXED_SURFACE.get(
-                    typeOfFirstFixedSurface, fixed_surface_missing
-                )
-                key = "scaleFactorOfFirstFixedSurface"
-                scaleFactorOfFirstFixedSurface = section[key]
-                typeOfSecondFixedSurface = section["typeOfSecondFixedSurface"]
-                if typeOfSecondFixedSurface != _TYPE_OF_FIXED_SURFACE_MISSING:
-                    if typeOfFirstFixedSurface != typeOfSecondFixedSurface:
-                        msg = (
-                            "Product definition section 4 has different "
-                            "types of first and second fixed surface"
-                        )
-                        raise TranslationError(msg)
```

Comment:
It looks like we're no longer checking this equality. Is this an oversight or deliberate?

Reply:
This is deliberate. The code now handles different types of fixed surface, and I have an example of this in a GRIB2 file (cloud fraction between land/sea surface and 1000 feet). Where different fixed surfaces cannot be coerced into one Coord, two Coords are applied to the cube.